### PR TITLE
t3c add support for anycast on http routed delivery services

### PR DESCRIPTION
**Pull request by jpappa200**
Closes: #7322

This PR provides support for anycast on edges with HTTP routed delivery services. It will add a remap line for each anycast partner for each http routed delivery service.

<hr/>

## Which Traffic Control components are affected by this PR?

- Documentation
- Traffic Control Cache Config (`t3c`, formerly ORT)

## What is the best way to verify this PR?

Configure 2 or more edges to share an anycast IP, queue updates and run t3c in syncds mode. verify in remap.config that http routed delivery services have a remap line for each anycast partner. 

## If this is a bugfix, which Traffic Control versions contained the bug?

Not a bugfix

## PR submission checklist
- [x] This PR has tests 
- [ ] This PR has documentation 
- [x] This PR has a CHANGELOG.md entry 
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://apache.org/security) for details)

### CODE REVIEW

File: lib/go-atscfg/remapdotconfig_test.go [hunk 1]
```diff
@@ -72,12 +91,203 @@ func pluginsFromTokens(tokens []string, prefix string) []string {
...
+		default:
+			t.Fatalf("unexpected remap line '%v'", line)
+		}
+	}
+	for key, val := range mappings {
+		if !val {
+			t.Fatalf("expected to find remap rule for '%v'", key)
+		}
+	}
+}
```
Comment (traeak) [inline lib/go-atscfg/remapdotconfig_test.go:282]: Forgot blank line before next function definition.

File: lib/go-atscfg/remapdotconfig_test.go [hunk 2]
```diff
@@ -72,12 +91,203 @@ func pluginsFromTokens(tokens []string, prefix string) []string {
...
 func TestMakeRemapDotConfig0(t *testing.T) {
 	hdr := "myHeaderComment"
 
 	server := makeTestRemapServer()
 	server.Type = "EDGE"
 
+	servers := makeTestAnyCastServers()
```
Comment (traeak) [inline lib/go-atscfg/remapdotconfig_test.go:289]: would it be better to make some/most of the normal tests use an empty collection for this?

File: lib/go-atscfg/remapdotconfig_test.go (hunk 1 above)
Comment (jpappa200) [reply lib/go-atscfg/remapdotconfig_test.go:282]: Line has been added

**Review by traeak — APPROVED** [review]
Looks good.  Tested with 3 servers with lo service address set.  Generated repeated remap.config entries for http services, also changed value of service ip on one which worked as expected.

File: CHANGELOG.md [hunk 3]
```diff
@@ -29,6 +29,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - [#7273](https://github.com/apache/trafficcontrol/pull/7273) *Traffic Ops* Adds SSL-KEY-EXPIRATION:READ permission to operations, portal, read-only, federation and steering roles
 - [#7296](https://github.com/apache/trafficcontrol/pull/7296) *Traffic Portal* New configuration option in `traffic_portal_properties.json` at `deliveryServices.exposeInactive` controls exposing APIv5 DS Active State options in the TP UI.
 - [#7332](https://github.com/apache/trafficcontrol/pull/7332) *Traffic Ops* Creates new role needed for TR to watch TO resources.
+- [#7309](https://github.com/apache/trafficcontrol/pull/7309) *Traffic Ops* Creates new role needed for TR to watch TO resources
```
Comment (ocket8888) [inline CHANGELOG.md:32]: #7309 was reverted; its changelog entry should remain absent

**Review by ocket8888 — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (jpappa200) [reply CHANGELOG.md]: removed 7309

**Review by ocket8888 — APPROVED** [review]

**Pull request merged**